ui_testing/testcases/header_tests/test08_usermanagement.py

```diff
@@ -238,4 +227,289 @@ def test010_update_user_role_with_save_cancel_btn(self, save):
+        self.assertTrue(result_user, user_filter)
+        self.base_selenium.click(element='user_management:filter_reset_btn')
+
+    def test014_filter_by_email(self):
+        """
+        User management Approach: I can filter by user email successfully
+        LIMS-6442
+        :return:
+        """
+        self.base_selenium.click(element='header:user_management_button')
+        random_user_name = self.generate_random_string()
+        random_email = self.header_page.generate_random_email()
+        self.header_page.create_new_user(user_name=random_user_name, user_email=random_email,
+                                         user_role='', user_password='1', user_confirm_password='1')
+
+        self.base_selenium.click(element='general:menu_filter_view')
+        user_filter = self.header_page.filter_user_by(filter_element='user_management:filter_email',
+                                                      filter_text=random_email)
+
+        result_user = self.header_page.result_table()[0]
+        self.assertTrue(result_user, user_filter)
+        self.base_selenium.click(element='user_management:filter_reset_btn')
+
+    def test015_filter_by_role(self):
+        """
+        User management Approach: I can filter by user role successfully
+        LIMS-6443
+        :return:
+        """
+        # create random role
+        self.base_selenium.click(element='header:roles_and_permissions_button')
+        random_role_name = self.generate_random_string()
+        self.header_page.create_new_role(role_name=random_role_name)
+
+        self.base_selenium.LOGGER.info('make sure that that the user record created in the active table')
+        created_role = self.header_page.search(random_role_name)[0]
+        role_data = self.base_selenium.get_row_cells_dict_related_to_header(row=created_role)
+        self.assertTrue(created_role, role_data)
+
+        # create user with this role to filter by it
+        self.header_page.click_on_header_button()
+        self.base_selenium.click(element='header:user_management_button')
+        random_user_name = self.generate_random_string()
+        random_user_email = self.base_page.generate_random_email()
+        self.header_page.create_new_user(user_name=random_user_name,user_email=random_user_email,
+                                         user_role=random_role_name, user_password='1', user_confirm_password='1')
+
+        self.base_selenium.click(element='general:menu_filter_view')
+        self.header_page.set_user_role(user_role=random_role_name)
+        user_role = self.header_page.get_user_role()
+
+        user_filter = self.header_page.filter_user_by(filter_element='user_management:filter_role',
+                                                      filter_text=user_role)
+
+        result_user = self.header_page.result_table()[0]
+        self.assertTrue(result_user, user_filter)
+        self.base_selenium.click(element='user_management:filter_reset_btn')
```

Comment:
duplicate with  the same test case?

Reply:
@0xIslamTaha this is not a duplicate because I want to create random role then random user by this role then filter by the user role it self so where is the duplication from this